src/Switch-Keithley_707B/main.py

```diff
@@ -34,158 +34,147 @@
-
-from ErrorMessage import error
+import re  # For regex / string validation
+import time
 
 from EmptyDeviceClass import EmptyDevice
-import time
-import re # For regex / string validation
 
-class Device(EmptyDevice):
 
+class Device(EmptyDevice):
     description = """
         <p><strong>Usage:</strong></p>
         <ul>
         <li>Enter comma separated configurations, e.g "1101; 1201; 1301".</li>
         <li>Only certain cards have a bank. For cards without a bank, just put 1 for the bank number.</li>
-        <li>Open means the relay is open/disconnected. No current will flow through. Closed means the relay is closed/connected, and current will flow through.</li>
-        <li>In the user manual (707B-901-01 Rev. B / January 2015) page 152 you can find how Matrix card channel specifiers are for this instrument.</li>
+        <li>Open means the relay is open/disconnected. No current will flow through. Closed means the relay is closed/
+        connected, and current will flow through.</li>
+        <li>In the user manual (707B-901-01 Rev. B / January 2015) page 152 you can find how Matrix card channel 
+        specifiers are for this instrument.</li>
 
         </ul>
         <p>&nbsp;</p>
         <p><strong>Known issues:</strong></p>
-        <p>In case SweepBox is used as Sweep value, comma-separated configurations can not be used at the moment as the comma is already used by the SweepBox to split the values to be set.</p>
+        <p>In case SweepBox is used as Sweep value, comma-separated configurations can not be used at the moment as the 
+        comma is already used by the SweepBox to split the values to be set.</p>
     """
 
-    def __init__(self):
-        
+    def __init__(self) -> None:
+        """Set up device parameters."""
         EmptyDevice.__init__(self)
-        
+
         self.shortname = "Keithley 707B"
-        
+
         self.port_manager = True
-        self.port_types = ['GPIB', 'TCPIP']
-        self.port_properties = {    
-                                    "timeout": 5,
-                                    "EOL": "\r",
-                                }
-        self.port_identifications = ['Keithley Instruments,707B', 'Keithley Instruments Inc., Model 707B']
-        self.switch_settling_time_s = 0.004                   
+        self.port_types = ["GPIB", "TCPIP"]
+        self.port_properties = {
+            "timeout": 5,
+            "EOL": "\r",
+        }
+        self.port_identifications = ["Keithley Instruments,707B", "Keithley Instruments Inc., Model 707B"]
 
         self.variables = ["Channels"]
         self.units = ["#"]
-        #self.plottype = [True]
-        #self.savetype = [True]
+        self.plottype = [False]
+        self.savetype = [True]
 
-    def set_GUIparameter(self):
-        
-        GUIparameter = {
+        # Measurement parameters
+        self.sweepmode: str = "Channels"
+        self.switch_settling_time_s: float = 0.004
+
+    @staticmethod
```

Comment:
set_GUIparameter should not be a staticmethod. Even though it does not need `self`, the parent class (EmptyDevice) defines set_GUIparameter to be a non-static method and child classes should use the same signature